impl/core/src/main/java/io/serverlessworkflow/impl/WorkflowApplication.java

```diff
@@ -141,7 +140,7 @@ public SchemaValidator getValidator(SchemaInline inline) {
     private ResourceLoaderFactory resourceLoaderFactory = DefaultResourceLoaderFactory.get();
     private SchemaValidatorFactory schemaValidatorFactory;
     private WorkflowPositionFactory positionFactory = () -> new QueueWorkflowPosition();
-    private WorkflowInstanceIdFactory idFactory = () -> UlidCreator.getMonotonicUlid().toString();
+    private WorkflowInstanceIdFactory idFactory = new MonotonicUlidWorkflowInstanceIdFactory();
```

Comment:
@mcruzdev In order to prevent SecureReandom instance to be created if the default id factory is overriden, this line neeeds to be moved to build() method